components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/UserAwareAPIConsumer.java

```diff
@@ -127,18 +127,6 @@ public Comment getComment(Identifier identifier, String commentId) throws APIMan
         return super.getComment(identifier, commentId);
     }
 
-    @Override
-    public Comment[] getComments(ApiTypeWrapper apiTypeWrapper) throws APIManagementException {
-        checkSubscribePermission();
-        return super.getComments(apiTypeWrapper);
-    }
-
-    @Override
-    public void deleteComment(APIIdentifier identifier, String commentId) throws APIManagementException {
-        checkSubscribePermission();
-        super.deleteComment(identifier,commentId);
-    }
-
```

Comment:
I don't think comment deletion should be allowed for an anonymous user

Reply:
Sorry, error on my end. Fixed now. Only getComment and getComments methods should be removed. Thanks for pointing out.